src/NewButton/types.ts

```diff
@@ -27,10 +31,28 @@ export type ButtonProps = {
    * The trailing icon comes after button content
    */
   trailingIcon?: React.FunctionComponent<IconProps>
+  children: React.ReactNode
+  as?: React.ElementType
```

Comment:
I shouldn't need to add this if I use `PolymorphicForwardRefComponent` but typescript simply wouldn't recognise this without explicit prop definition.